File: sdks/ruby/lib/novabilling/types/applied_add_on_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class AppliedAddOnResponse < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :add_on_id, -> { String }, optional: false, nullable: false, api_name: "addOnId"
      field :customer_id, -> { String }, optional: false, nullable: false, api_name: "customerId"
      field :subscription_id, -> { String }, optional: true, nullable: false, api_name: "subscriptionId"
      field :amount, -> { String }, optional: false, nullable: false
      field :currency, -> { String }, optional: false, nullable: false
      field :invoice_id, -> { String }, optional: true, nullable: false, api_name: "invoiceId"
      field :created_at, -> { String }, optional: false, nullable: false, api_name: "createdAt"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/AppliedAddOnResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record AppliedAddOnResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("addOnId")]
    public required string AddOnId { get; set; }

    [JsonPropertyName("customerId")]
    public required string CustomerId { get; set; }

    [JsonPropertyName("subscriptionId")]
    public string? SubscriptionId { get; set; }

    [JsonPropertyName("amount")]
    public required string Amount { get; set; }

    [JsonPropertyName("currency")]
    public required string Currency { get; set; }

    [JsonPropertyName("invoiceId")]
    public string? InvoiceId { get; set; }

    [JsonPropertyName("createdAt")]
    public required string CreatedAt { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
